tests/unit/test_models.py

```diff
@@ -50,5 +55,78 @@ def test_invalid(self):
             TransferType.from_str(self.invalid_type)
 
 
+class TestTransfer(unittest.TestCase):
+
+    def test_add_slippage(self):
+        solver = Address.zero()
+        transfer = Transfer(
+            token_type=TokenType.NATIVE,
+            token_address=None,
+            receiver=solver,
+            amount=1.0
+        )
+        positive_slippage = SolverSlippage(
+            solver_name="Test Solver",
+            solver_address=solver,
+            amount_wei=10 ** 18 // 2
+        )
+        negative_slippage = SolverSlippage(
+            solver_name="Test Solver",
+            solver_address=solver,
+            amount_wei=-10 ** 18 // 2
+        )
+        transfer.add_slippage(positive_slippage)
+        self.assertAlmostEqual(transfer.amount, 1.5, delta=0.0000000001)
+        transfer.add_slippage(negative_slippage)
+        self.assertAlmostEqual(transfer.amount, 1.0, delta=0.0000000001)
```

Comment:
Note: I think one test is missing:
You do not test for:
Invalid adjustment

Not sure how important it is

Reply:
Right definitely. Great point.

Reply:
Added the missing test. Definitely important to have.